lib/rubocop/cop/layout/comment_indentation.rb

```diff
@@ -154,7 +154,9 @@ def correct_indentation(next_line)
         end
 
         def less_indented?(line)
-          /^\s*(end\b|[)}\]])/.match?(line)
+          rule = config.for_cop('Layout/AccessModifierIndentation')['EnforcedStyle'] == 'outdent'
+          access_modifier = 'private|protected|public'
+          /\A\s*(end\b|[)}\]])/.match?(line) || (rule && /\A\s*(#{access_modifier})\b/.match?(line))
```

Comment:
I would suggest to extract `/\A\s*(#{access_modifier})\b/` into a constant or inline `access_modifier` into it, because this method call will always generate unnecessary allocations (needs measuring to know the impact).